docs/reference/contributing/guidelines/workflow.md

```diff
@@ -80,7 +88,43 @@ Each feature has a tech lead. This person is responsible for:
 - Rebasing often to track master development.
 - Reviewing any addition to the feature branch (approval required by the feature tech lead or another assigned person).
 
-#### Pull request template
+##### Pull request types
+
+We consider the following pull request types.
+
+##### Fix
+
+A bug fix is backward-compatible internal change that fixes incorrect behavior.
+
+Release: patch
+
+##### Refactor
+
+Refactors are intended for feature releases if they are not fixing specific issue as they can introduce new issues.
+
+Release: feature
+
+##### New target
+
+Adding a new target targets the patch release as it updating targets folder implementation.
+
+Release: patch
+
+##### Feature
+
+New features targets feature releases. It can be integrated only if the feature supports most of the targets (if it requires new target HAL implementation).
+
+Adding a new functionality is considered to be a feature. It does not matter if it is C++, C or Python.
+
+Release: feature
+
+##### Breaking change
+
+Any change that results in breaking user space. It should have strong justification to be considered. Often the change could be made backward compatible, for example deprecate the old functionality and introduce the new replacement.
```

Comment:
@SenRamakri asked good question. As this is quite serious type of pull request, we might want to clarify more.

Shall I add `breaking user space (an application requires changes to continue working)` or similar.  This also leads to another good question is - what can actually break (what is considered public vs internal) ?